### ISSUE TITLE
C or CPP doesn’t support socket programming?

### ISSUE THREAD

**Issue by stan12138** [holzschu/a-shell#65]
When compile c socket script, errors happened:

server.c:23:11: warning: implicit declaration of function 'socket' is invalid in C99 [-Wimplicit-function-declaration]
        server = socket(AF_INET, SOCK_STREAM, 0);
                 ^
server.c:29:2: warning: implicit declaration of function 'bind' is invalid in C99 [-Wimplicit-function-declaration]
        bind(server, (struct sockaddr *)&server_address, sizeof(server_address));
        ^
server.c:30:2: warning: implicit declaration of function 'listen' is invalid in C99 [-Wimplicit-function-declaration]
        listen(server, 20);
        ^
server.c:37:12: warning: implicit declaration of function 'accept' is invalid in C99 [-Wimplicit-function-declaration]
                client = accept(server, (struct sockaddr *)&client_address, &client_address_len);

**Comment by holzschu** [comment]
Yes, webAssembly, or rather WASI, does not have sockets at the moment. 

It is heavily discussed in WASI; I have no idea if I could add it to a-Shell's extensions of WASI, like I did with file operations. There would be a performance loss, as all communications from webAssembly to the outside world are making a trip webAssembly -> JavaScript -> iOS -> JavaScript -> webAssembly.

**Comment by stan12138** [comment]
Ok, thanks for the answer and your app.
